lib/selinux.c

```diff
@@ -98,6 +99,146 @@ int reset_selinux_file_context (void)
 	return 0;
 }
 
+/*
+ *  Log callback for libselinux internal error reporting.
+ */
+__attribute__((__format__ (printf, 2, 3)))
+static int selinux_log_cb (int type, const char *fmt, ...) {
+	va_list ap;
+	char *buf;
+	int r;
+#ifdef WITH_AUDIT
+	static int selinux_audit_fd = -2;
+#endif
+
+	va_start (ap, fmt);
+	r = vasprintf (&buf, fmt, ap);
+	va_end (ap);
+
+	if (r < 0) {
+		free (buf);
```

Comment:
buf could be uninitialized here?

Reply:
fixed, thanks

Reply:
Still not sure it is safe/necessary to free(buf) here.  man page for vasprintf() says "If memory allocation wasn't  possible,  or  some other error occurs, these functions will return -1, and the contents of strp are undefined."  That means buf could be undefined after the call if it returns -1.

Reply:
you're absolutely right, fixed; thanks